internal/aws/containerinsight/utils.go

```diff
@@ -259,27 +258,35 @@ func ConvertToOTLPMetrics(fields map[string]any, tags map[string]string, logger
 	for key, value := range fields {
 		metric := RemovePrefix(metricType, key)
 		unit := GetUnitForMetric(metric)
+		scopeMetric := ilms.AppendEmpty()
 		switch t := value.(type) {
 		case int:
-			intGauge(ilms.AppendEmpty(), key, unit, int64(t), timestamp)
+			intGauge(scopeMetric, key, unit, int64(t), timestamp)
 		case int32:
-			intGauge(ilms.AppendEmpty(), key, unit, int64(t), timestamp)
+			intGauge(scopeMetric, key, unit, int64(t), timestamp)
 		case int64:
-			intGauge(ilms.AppendEmpty(), key, unit, t, timestamp)
+			intGauge(scopeMetric, key, unit, t, timestamp)
 		case uint:
-			intGauge(ilms.AppendEmpty(), key, unit, int64(t), timestamp)
+			intGauge(scopeMetric, key, unit, int64(t), timestamp)
 		case uint32:
-			intGauge(ilms.AppendEmpty(), key, unit, int64(t), timestamp)
+			intGauge(scopeMetric, key, unit, int64(t), timestamp)
 		case uint64:
-			intGauge(ilms.AppendEmpty(), key, unit, int64(t), timestamp)
+			intGauge(scopeMetric, key, unit, int64(t), timestamp)
 		case float32:
-			doubleGauge(ilms.AppendEmpty(), key, unit, float64(t), timestamp)
+			doubleGauge(scopeMetric, key, unit, float64(t), timestamp)
 		case float64:
-			doubleGauge(ilms.AppendEmpty(), key, unit, t, timestamp)
+			doubleGauge(scopeMetric, key, unit, t, timestamp)
 		default:
 			valueType := fmt.Sprintf("%T", value)
 			logger.Warn("Detected unexpected field", zap.String("key", key), zap.Any("value", value), zap.String("value type", valueType))
 		}
+		if addAttributesAtDatapoints {
```

Comment:
Yup, EMV export does flatten attributes. This is to pass down resource attributes down to datapoints so that GPU attribute processor in the agent can run some logics (eg filtering) based on these.

Reply:
Can that logic in the GPU attribute processor be updated to look at both resource and datapoint attrs? Trying to avoid an increase in memory.